app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/VisibleMessageView.kt

```diff
@@ -243,44 +243,125 @@ class VisibleMessageView : LinearLayout {
+        //   i.) incoming AND scheduled to disappear.
+        //   ii.) outgoing but NOT scheduled to disappear, or
+        //   iii.) outgoing AND scheduled to disappear.
+
+        // ----- Case i..) Message is incoming and scheduled to disappear -----
+        if (message.isIncoming && scheduledToDisappear) {
+            binding.messageStatusTextView.isVisible  = true  // Display the status ('Read')..
+            binding.messageStatusImageView.isVisible = false // ..but don't display the status image..
+            binding.expirationTimerView.isVisible    = true  // ..and instead show the timer.
+            binding.expirationTimerView.bringToFront()
+            updateExpirationTimer(message)
+            return
+        }
+
+        // --- If we got here then we know the message is outgoing ---
 
-            // Always show the delivery status of the last sent message
-            val thisUsersSessionId = TextSecurePreferences.getLocalNumber(context)
-            val lastSentMessageId = mmsSmsDb.getLastSentMessageFromSender(message.threadId, thisUsersSessionId)
-            val isLastSentMessage = lastSentMessageId == message.id
+        val thisUsersSessionId = TextSecurePreferences.getLocalNumber(context)
+        val lastSentMessageId = mmsSmsDb.getLastSentMessageFromSender(message.threadId, thisUsersSessionId)
+        val isLastSentMessage = lastSentMessageId == message.id
 
-            binding.messageStatusTextView.isVisible = textId != null && (isLastSentMessage || scheduledToDisappear)
-            val showTimer = scheduledToDisappear && !message.isPending
-            binding.messageStatusImageView.isVisible = iconID != null && !showTimer && (!message.isSent || isLastSentMessage)
+        // ----- Case ii.) Message is outgoing but NOT scheduled to disappear -----
+        if (!scheduledToDisappear) {
+            // If this isn't a disappearing message then we never show the timer
+            binding.expirationTimerView.isVisible = false
 
-            binding.messageStatusImageView.bringToFront()
-            binding.expirationTimerView.bringToFront()
-            binding.expirationTimerView.isVisible = showTimer
-            if (showTimer) updateExpirationTimer(message)
-        } else {
-            binding.messageStatusTextView.isVisible = false
-            binding.messageStatusImageView.isVisible = false
+            // If the message has NOT been successfully sent then always show the delivery status text and image..
+            val neitherSentNorRead = !(message.isSent || message.isRead)
+            if (neitherSentNorRead) {
+                binding.messageStatusTextView.isVisible = true
+                binding.messageStatusImageView.isVisible = true
+            }
+            else // ..but if the message HAS been successfully sent or read..
+            {
+                // ..then we'll only display the delivery status text and image if this is the last sent message..
```

Comment:
We were supposed to show the status text if there is a timer.

So we always show status text an an icon/timer together.

Reply:
Yeah, fair enough - just confirmed w/ Morgan and checked that desktop keeps the text as well (which it does).